modules/libcom/src/osi/os/WIN32/epicsAtomicOSD.h

```diff
@@ -19,10 +19,10 @@
 
 #define EPICS_ATOMIC_OS_NAME "WIN32"
 
-#ifdef VC_EXTRALEAN
-#   define VC_EXTRALEAN_DETECTED_epicsAtomicOSD_h
+#ifdef WIN32_LEAN_AND_MEAN
+#   define WIN32_LEAN_AND_MEAN_DETECTED_epicsAtomicOSD_h
 #else
-#   define VC_EXTRALEAN
+#   define WIN32_LEAN_AND_MEAN
 #endif
```

Comment:
It would be nice to include a brief description of what all this macro magic achieves as a comment towards the top of the file, for future maintainers to read without having to research for themselves.

I'm wondering whether all the `_DETECTED_epicsAtomicOSD_h` should be undefined at the end or not; this header does have a `#include` guard, so later includes of the same file won't ever see them and nothing else in Base is using them AFAICT. Not a big point, just a minor clean-up if you're making the above change.

Reply:
I've added a couple comments about `WIN32_LEAN_AND_MEAN` and `NOMINMAX`

Just want to note that [push_macro](https://gcc.gnu.org/onlinedocs/gcc/Push_002fPop-Macro-Pragmas.html) may be a less-verbose solution to saving macro definitions in the future. Minimum GCC version for this is 4.2.0, Clang has supported it since 2010 and MSVC added it before GCC. But I think the vxWorks compilers are still too old for this feature.

Reply:
> ... vxWorks compilers are still too old for this feature.

Which is relevant in a `os/WIN32/` header how?  ;)

@FreddieAkeroyd Any objections to usage of `push_macro()` ?

Reply:
> Which is relevant in a os/WIN32/ header how? ;)

Eventually I'll learn to have my morning coffee *before* making comments on my pull requests :laughing:

Reply:
`push_macro` seems a good approach to improve readability

Reply:
Ok, I've switched this to use `push_macro`/`pop_macro`. I also added some additional `NOXXX` defines to strip out as much as possible. Seems like `WIN32_LEAN_AND_MEAN` covers most.